components/backup-stream/src/endpoint.rs

```diff
@@ -219,7 +219,7 @@ where
         self.meta_client.clone()
     }
 
-    fn on_fatal_error_of_task(&self, task: &str, err: &Error) -> future![()] {
+    fn on_fatal_error_of_task(&self, task: &str, err: &Error) -> future![bool] {
```

Comment:
since we are modifying this part, can we add a fatal/error log in this method, I noticed that it only uploads the status but in the log we don't see the error. It could be easier when looking at TiKV logs searching for error keyword.

Reply:
When receiving a fatal error message, the caller may call `err.report_fatal()`:
<img width="735" alt="image" src="https://github.com/user-attachments/assets/306bf5e7-ebb3-4b17-8d96-44e182201366" />

Reply:
there are two call sites if I'm not on a really old master, one is at the `register_stream_task` and I don't find any logging associated with it.
<img width="709" alt="image" src="https://github.com/user-attachments/assets/093858cc-bebe-4824-9068-6dd58de1e2d0" />
I feel like it would be better to log where we handle the error, it could be easier to maintain and we don't need to keep in mind to log it every time we call the on_fatal_error